### PR TITLE
util/groupmember: remove redundant code

### LINKED ISSUE #3001: util/groupmember: delete package once we use Go 1.18
The util/groupmember package will become redundant once we have:

https://github.com/golang/go/commit/30faf968b1f348e944db3bde24c13462125007b1

... in our tree.

Either via Go 1.18 or cherry-picking it into tailscale/go.

/cc @maisem @josharian

### PR DESCRIPTION
**Pull request by maisem**
Now that we have https://github.com/golang/go/commit/30faf968b1f348e944db3bde24c13462125007b1
this is no longer needed.

Fixes #3001

Signed-off-by: Maisem Ali <maisem@tailscale.com>

### CODE REVIEW

File: util/groupmember/groupmember.go [hunk 1]
```diff
@@ -7,14 +7,44 @@
...
+	}
+	for _, ugid := range ugids {
+		if gid == ugid {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+var groupIDCache sync.Map // of string
```
Comment (bradfitz) [inline util/groupmember/groupmember.go:37]: // from string (group name) to string (Gid)

File: util/groupmember/groupmember.go [hunk 2]
```diff
@@ -7,14 +7,44 @@
...
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+var groupIDCache sync.Map // from string (group name) to string (Gid)
+
+func getGroupID(groupName string) (string, error) {
+	s, ok := groupIDCache.Load(groupName)
```
Comment (bradfitz) [inline util/groupmember/groupmember.go:40]: caches for security stuff like this scare me a bit.

also, a more common pattern I think is:

      if s, ok := groupIDCache.Load(groupName).(string); ok {
             return s, nil
       }

That's safe even if somebody puts the wrong value type in that key.

Comment (bradfitz) [reply util/groupmember/groupmember.go:40]: But I guess it was already caching, so sure?

**Review by bradfitz — APPROVED** [review]

File: util/groupmember/groupmember.go (hunk 2 above)
Comment (maisem) [reply util/groupmember/groupmember.go:40]: I can just drop the cache

Comment (maisem) [reply util/groupmember/groupmember.go]: Dropped the cache

Comment (maisem) [reply util/groupmember/groupmember.go]: PTAL

Comment (bradfitz) [reply util/groupmember/groupmember.go]: Either way. Or a TODO to investigate whether it matters.

In any case, the cache should probably be up a level on the func itself (is this user in this group?) and only cache positive caching for 10 seconds or so.

**Review by bradfitz — APPROVED** [review]

**Pull request merged**